### PR TITLE
feat: add `noMisalignedNullability()` rule

### LINKED ISSUE #3: Check the nullability alignment with JPA annotations
When a field or method is annotated with JPA schema annotations, check if a corresponding nullability annotation is present that aligns with the nullability of the entity definition.

The MVP implementation may only contain checks for annotations present on fields.

### PR DESCRIPTION
**Pull request by Charm-Xin**
> I still need to figure out how to name the methods and describe the rule.

My thought process is this:

- If the property is nullable or optional, then it must be annotated with jetbrains [Nullable]

### CODE REVIEW

File: src/main/kotlin/dev/ocpd/jsensible/rules/jpa/JpaRules.kt [hunk 1]
```diff
@@ -28,4 +32,10 @@ object JpaRules {
     fun noEagerFetch(): ArchRule =
         noMembers().should(useEagerFetch())
             .because("no property should be fetched eagerly by default")
+
+    fun noNullableOrOptional(): ArchRule =
```
Comment (unlimitedsola) [inline src/main/kotlin/dev/ocpd/jsensible/rules/jpa/JpaRules.kt]:
```suggestion
    fun noMisalignedNullability(): ArchRule =
```
I think you got the implementation right but the naming is a bit confusing across the board. As we do have `Optional<T>` in JDK so naming it this way is a bit ambiguous.

**Review by unlimitedsola — COMMENTED** [review]
Could you also provide a test for this rule in this PR?

**Comment by Charm-Xin** [comment]
> Could you also provide a test for this rule in this PR?

sure.

**Comment by Charm-Xin** [comment]
close https://github.com/ocpddev/jsensible/issues/3

**Review by unlimitedsola — APPROVED** [review]

**Pull request merged**